server/src/internalClusterTest/java/org/opensearch/wlm/WorkloadManagementStatsIT.java

```diff
@@ -0,0 +1,315 @@
+        QueryGroup queryGroup = new QueryGroup(
+            NAME1,
+            new MutableQueryGroupFragment(MutableQueryGroupFragment.ResiliencyMode.ENFORCED, Map.of(ResourceType.CPU, 0.5))
+        );
+        String id = queryGroup.get_id();
+        updateQueryGroupInClusterState(PUT, queryGroup);
+        WlmStatsResponse response = getWlmStatsResponse(null, new String[] { _ALL }, true);
+        validateResponse(response, null, new String[] { DEFAULT_QUERY_GROUP, id });
+
+        WlmStatsResponse response2 = getWlmStatsResponse(null, new String[] { _ALL }, false);
+        validateResponse(response2, new String[] { DEFAULT_QUERY_GROUP, id }, null);
+
+        WlmStatsResponse response3 = getWlmStatsResponse(null, new String[] { _ALL }, null);
+        validateResponse(response3, new String[] { DEFAULT_QUERY_GROUP, id }, null);
+
+        updateQueryGroupInClusterState(DELETE, queryGroup);
+    }
+
+    public void testWlmStatsWithNodesId() throws Exception {
+        QueryGroup queryGroup = new QueryGroup(
+            NAME1,
+            new MutableQueryGroupFragment(MutableQueryGroupFragment.ResiliencyMode.ENFORCED, Map.of(ResourceType.CPU, 0.5))
+        );
+        String queryGroupId = queryGroup.get_id();
+        String nodeId = client().admin().cluster().prepareState().clear().setNodes(true).get().getState().nodes().getLocalNodeId();
+        updateQueryGroupInClusterState(PUT, queryGroup);
+        WlmStatsResponse response = getWlmStatsResponse(new String[] { nodeId }, new String[] { _ALL }, true);
+        validateResponse(response, new String[] { nodeId }, new String[] { DEFAULT_QUERY_GROUP, queryGroupId });
+
+        WlmStatsResponse response2 = getWlmStatsResponse(new String[] { nodeId, INVALID_ID }, new String[] { _ALL }, false);
+        validateResponse(response2, new String[] { nodeId, DEFAULT_QUERY_GROUP, queryGroupId }, new String[] { INVALID_ID });
+
+        WlmStatsResponse response3 = getWlmStatsResponse(new String[] { INVALID_ID }, new String[] { _ALL }, false);
+        validateResponse(response3, null, new String[] { nodeId, DEFAULT_QUERY_GROUP, queryGroupId });
+
+        updateQueryGroupInClusterState(DELETE, queryGroup);
+    }
+
+    public void testWlmStatsWithIdAndBreach() throws Exception {
+        QueryGroup queryGroup = new QueryGroup(
+            NAME1,
+            new MutableQueryGroupFragment(MutableQueryGroupFragment.ResiliencyMode.ENFORCED, Map.of(ResourceType.CPU, 0.5))
+        );
+        String queryGroupId = queryGroup.get_id();
+        String nodeId = client().admin().cluster().prepareState().clear().setNodes(true).get().getState().nodes().getLocalNodeId();
+        updateQueryGroupInClusterState(PUT, queryGroup);
+        WlmStatsResponse response = getWlmStatsResponse(new String[] { nodeId }, new String[] { DEFAULT_QUERY_GROUP }, true);
+        validateResponse(response, new String[] { nodeId }, new String[] { DEFAULT_QUERY_GROUP, queryGroupId });
```

Comment:
I am not able to see how we are making the certain query_group is breaching its thresholds ?